vulnerabilities/package_managers.py

```diff
@@ -321,3 +321,24 @@ async def fetch(self, owner_repo: str, session) -> None:
         resp = await resp.json()
         print(resp)
         self.cache[owner_repo] = [release["ref"].split("/")[-1] for release in resp]
+
+
+class HexVersionAPI(VersionAPI):
+    async def load_api(self, pkg_set):
+        async with ClientSession(raise_for_status=True) as session:
+            await asyncio.gather(
+                *[self.fetch(pkg, session) for pkg in pkg_set if pkg not in self.cache]
+            )
+
+    async def fetch(self, pkg, session):
+        url = f"https://hex.pm/api/packages/{pkg}"
+        versions = set()
+        try:
+            response = await session.request(method="GET", url=url)
+            response = await response.json()
+            for release in response["releases"]:
+                versions.add(release["version"])
+        except (ClientResponseError, JSONDecodeError):
```

Comment:
Do these get triggered ?

Reply:
I think these may get triggered if <code>session.request</code> or <code>response.json</code> fail. Actually these were used in other APIs so i added it. I can remove if you suggest.

Reply:
Don't remove it, it's better to be defensive here. 

I was asking since there are special cases where these fail, and it is a good practice to mention these in a comment .